db/db_impl_write.cc

```diff
@@ -180,6 +206,9 @@ Status DBImpl::WriteImpl(const WriteOptions& write_options,
     // that nobody else can be writing to these particular stats.
     // We're optimistic, updating the stats before we successfully
     // commit.  That lets us release our leader status early.
+    if (!batch_writes) {
+      stat_mutex_.Lock();
```

Comment:
AddDBStats is not atomic.
```
275   void AddDBStats(InternalDBStatsType type, uint64_t value) {
276     auto& v = db_stats_[type];
277     v.store(v.load(std::memory_order_relaxed) + value,
278             std::memory_order_relaxed);
279   }
```
StatisticsImpl::recordTick seems to be fine (using fetch_add) and PERF_TIMER_* seem to be thread-local and thus ok.